test/gtest/ucp/test_ucp_atomic.cc

```diff
@@ -124,17 +131,29 @@ class test_ucp_atomic : public test_ucp_memheap {
 
     virtual void init() {
         int atomic_mode = get_variant_value() & ATOMIC_MODE;
+        int odp         = get_variant_value() & ODP;
         const char *atomic_mode_cfg =
                         (atomic_mode == UCP_ATOMIC_MODE_CPU)    ? "cpu" :
                         (atomic_mode == UCP_ATOMIC_MODE_DEVICE) ? "device" :
                         (atomic_mode == UCP_ATOMIC_MODE_GUESS)  ? "guess" :
                         "";
         modify_config("ATOMIC_MODE", atomic_mode_cfg);
+
+        if (odp) {
+            modify_config("REG_NONBLOCK_MEM_TYPES", "host");
+            modify_config("IB_ODP_MEM_TYPES", "host", SETENV_IF_NOT_EXIST);
+            modify_config("IB_MLX5_DEVX_OBJECTS", "", SETENV_IF_NOT_EXIST);
+        }
         test_ucp_memheap::init();
+
+        if (odp && !(check_amo_odp_supported(sender(), UCS_MEMORY_TYPE_HOST))) {
+            cleanup();
```

Comment:
what is not cleaned up by the destructor?

Reply:
maybe this warning (warn_existing_eps) should be moved to test_base::TearDownProxy, after calling cleanup:
https://github.com/openucx/ucx/blob/97b551b607ff7adfeadc0c2367a9688e5d95438f/test/gtest/common/test.cc#L345-L347

Reply:
decided to keep it as other test use same method, and fix all in another PR